### PR TITLE
build: Clean up Python config

### PR DESCRIPTION
**Pull request by j-s-ashley**
Addresses [pyproject.toml](https://github.com/thaler-lab/Wasserstein/blob/master/pyproject.toml) inefficiencies mentioned in #10.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
If this gets rebased on `master` then this should be ready to go in I think.

File: pyproject.toml
Comment (matthewfeickert) [inline pyproject.toml:16]: Is this just a comment to self? Or is there a follow up action needed here? If just a comment and nothing needs to get edited in terms of files then maybe just remove this so that this doesn't get forgotten, as the Issue will get closed when resolved.

```suggestion
```

**Review by matthewfeickert — APPROVED** [review]

File: pyproject.toml
Comment (j-s-ashley) [reply pyproject.toml:16]: @henryiii mentioned in #10 that the custom architecture mentioned here will not be needed once MacOS-14 is supported. I don't know if removing it now will break things.

Comment (matthewfeickert) [reply pyproject.toml:16]: Ah 

> The custom arch stuff in pyproject.toml can then be removed, as native will work fine.

okay. Thanks I had missed that earlier. I think it is probably fine to delete that here too, as when you add in `macos-14` support this won't be needed anyway.

Comment (henryiii) [reply pyproject.toml]: Shouldn’t you do this when you add support for macos-14? With this present, either image will build the native architecture, and cross compile the other architecture. Without it, they will only build their own native architectures.

**Review by matthewfeickert — APPROVED** [review]
LGTM 👍

File: pyproject.toml
Comment (matthewfeickert) [reply pyproject.toml]: Sorry missed this as on mobile. Yes, but that seemed fine to lose cross compiling now as I was going to suggest that once things are building as expected with libomp to overhaul the wheel building CI to basically be a copy of boost-histogram's with modifications and so all of this would get added back there. If I am missing something here let me know.

Comment (j-s-ashley) [reply pyproject.toml:16]: I'll wait on making further changes until tomorrow to see what you two land on here.

Comment (matthewfeickert) [reply pyproject.toml:16]: @j-s-ashley I'd defer to @henryiii here, so leave this in for now. I think this is good to go from here.

**Pull request merged**
